x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmConfig.java

```diff
@@ -24,11 +24,24 @@
     private final ThreadContext threadContext;
 
     public RealmConfig(RealmIdentifier identifier, Settings settings, Environment env, ThreadContext threadContext) {
+        this(identifier, settings, env, threadContext, null);
+    }
+
+    public RealmConfig(RealmIdentifier identifier, Settings settings, Environment env, ThreadContext threadContext, Integer order) {
```

Comment:
I think it would be better for this constructor to be  protected, and update the tests to pass through real `Settings` object with an order config.

I know that's going to be a pain - so please feel free to convince me if you feel otherwise.

Reply:
Are you suggesting the caller should create a `Settings` object that actually includes an `order`? In this case, this constructor is no longer necessary? Am I getting this right?

Reply:
I initially actual prefer to **not** adding another constructor. But the number of call site changes made me change my mind. But I am happy to drop the constructor and make more call site changes if you favour this approach. Please let me know. Thanks

Reply:
Changed to `int`.